Drivers/Src/Minions.c

```diff
@@ -1,406 +1,27 @@
     }
 }
 
-/**
- * @brief   Read toggle bitmap
- * @return  returns uint8_t bitmap for toggle (LEFT/RIGHT)
- */
-uint8_t Lights_Toggle_Bitmap_Read(void) {
-    return lightToggleBitmap;
+bool Minion_Read_Switch(MinionPin_t pin){
+    if(pin == BRAKELIGHT){
+        return false; //trying to read from an output pin, can't do that.
+    }    
 
+    return (bool)BSP_GPIO_Read_Pin(PINS_LOOKARR[pin].port, PINS_LOOKARR[pin].pinmask);
 }
 
-/**
- * @brief   Set light to given state
- * @param   light Which light to set
- * @param   state What state to set the light to
- * @return  void
- */
-void Lights_Set(light_t light, State state) {
-    CPU_TS timestamp;
-    OS_ERR err;
-    State lightCurrentState = Lights_Read(light);
-    
-    if(lightCurrentState != state){     // Check if state has changed
-        uint8_t lightNewStates = lightStatesBitmap;
-    
-        lightNewStates &= ~(0x01 << light); // Clear bit corresponding to pin
-        lightNewStates |= (state << light); // Set value to inputted state   
-
-        // Initialize tx buffer and gpio port b
-        uint8_t txWriteBuf[3] = {SPI_OPCODE_W, SPI_GPIOB, 0x00};
-        
-        if (light == BrakeLight) {  // Brakelight is only external
-            BSP_GPIO_Write_Pin(LIGHTS_PORT, BRAKELIGHT_PIN, !state);    // Negative logic
-            lightStatesBitmap = lightNewStates;   // Update lights bitmap
-            return;
-        } else {
-            txWriteBuf[2] = lightNewStates ^ 0x18;  // Flip left and right blink for negative logic on minion board
-
-            // Write to port c for lights present externally
-            switch (light) {
-                case LEFT_BLINK:
-                    BSP_GPIO_Write_Pin(LIGHTS_PORT, LEFT_BLINK_PIN, !state);
-                    break;
-                case RIGHT_BLINK:
-                    BSP_GPIO_Write_Pin(LIGHTS_PORT, RIGHT_BLINK_PIN, !state);
-                    break;
-                case Headlight_ON:
-                    BSP_GPIO_Write_Pin(LIGHTS_PORT, HEADLIGHT_PIN, !state);
-                    break;
-                default:
-                    break;
-            }
-        }
-        OSMutexPend(
-            &CommMutex,
-            0, 
-            OS_OPT_PEND_BLOCKING, 
-            &timestamp, 
-            &err);    // Lock mutex in order to update our lights bitmap and write to SPI
-        assertOSError(OS_BLINK_LIGHTS_LOC, err);
-
-        // Write to GPIOB on the minion board (SPI) for internal lights
-        ChipSelect();
-        BSP_SPI_Write(txWriteBuf, 3);
-        ChipDeselect();
-        
-        lightStatesBitmap = lightNewStates;   // Update lights bitmap
-
-        OSMutexPost(
-            &CommMutex,
-            OS_OPT_POST_NONE,
-            &err); // Unlock mutex
-        assertOSError(OS_BLINK_LIGHTS_LOC, err);
-    }
+void Minion_Write_Brakelight(State state){
+    BSP_GPIO_Write_Pin(PINS_LOOKARR[BRAKELIGHT].port, PINS_LOOKARR[BRAKELIGHT].pinmask, state);
 }
```

Comment:
Same idea as my earlier comment. This isn't generic behavior. I would suggest having a generic `Minion_Write_Output` function that can take in any pin, and any state, and set it appropriately.